File: Sorting/InsertionSort.java
// Name : Anish Sharma
// Enrollment No. : 202203103510042

package Sorting;

public class InsertionSort {
    static void insertionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int j = i;
            while (j > 0 && arr[j-1] > arr[j]) {
                int temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp;
                j--;
            }
        }
    }
    public static void main(String[] args) {
        int[] arr = {1,5,8,6,8,62,2,59,52,5,98,95,6,49,0,-65,-1};
        
        System.out.println("\n---Insertion Sort---\n");
        System.out.print("Array  : ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
        insertionSort(arr);

        System.out.print("Sorted : ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}

File: Sorting/InsertionSort.cpp
#include<iostream>
using namespace std;

void InsertionSort(int arr[], int n) {
    for (int i = 0; i < n; i++)
    {
        int j = i;
        while (j > 0 && arr[j-1] >= arr[j])
        {
            int temp = arr[j];
            arr[j] = arr[j-1];
            arr[j-1] = temp;
            j--;
        }
    }
} 

int main(int argc, char const *argv[])
{
    int arr[] = {1,5,6,8,5,96,2,54,952,58,475,54,-9875,-23};
    int count = sizeof(arr)/sizeof(arr[0]);
    InsertionSort(arr, count);

    for (int i = 0; i < count; i++)
    {
        cout << arr[i] << " ";
    }
    return 0;
}